src/probabilities_estimators/dispersion/dispersion.jl

```diff
@@ -98,7 +98,8 @@ function embed_symbols(symbols::AbstractVector, m, τ)
 end
 
 function dispersion_histogram(x::AbstractDataset, N, m, τ)
-    return fasthist(x.data, (N - (m - 1)*τ))
+
+    return fasthist(x.data) #n_tot (N - (m - 1)*τ)
```

Comment:
Why do you pass `x.data` instead of `x` and what does this comment mean? is it the total amount of points in `x`? If so, why is it necessary? And also, why does this function take in 4 arguments but only use 1?

Reply:
> Why do you pass x.data instead of x and what does this comment mean? is it the total amount of points in x? If so, why is it necessary? And also, why does this function take in 4 arguments but only use 1?

The commented term is supposed to normalize the histogram to a custom number of elements. Because `fasthist` always returns `Vector{Int}`, this requires conversion to floats. Check out my latest commit to see if the fix, and the adjustment to `fasthist` is satisfactory (avoids unnecessary conversions, because you can specify return type for the histogram)

Reply:
Returning `x.data` had something to do with the fact that in the old code, `Dataset`s couldn't be sorted, but `Vector{SVector}` could. We can simply pass `x` here now.

Reply:
No, I don't think this adjustment is appropriate. You anyways convert everything to float when you do `fasthist() ./ N` so I'm not sure why you care about the return type of `fasthist` in the first place.

Reply:
> Returning x.data had something to do with the fact that in the old code, Datasets couldn't be sorted, but Vector{SVector} could. We can simply pass x here now.

Actually, we can't simply pass `x`, because `Dataset`s cannot be sorted.

> I'm not sure why you care about the return type of fasthist in the first place

Because if the histogram already consists of floats (as determined by return type), no conversions to float are needed in `dispersion_histogram` (if dividing by a float). However, I'm not sure what the performance benefit is (if any).

Reply:
Trying

```julia
using DelayEmbeddings, Entropies
D = Dataset([rand(1:10, 3) for i = 1:100])
@test Entropies.fasthist(D) isa Vector{Int}
```

gives

```julia
  MethodError: no method matching sort!(::Dataset{3, Int64}; alg=Base.Sort.QuickSortAlg())
```

Reply:
Anyways, I have to continue this tomorrow. If you're still continuing, feel free to reverse the change to `fasthist`. Otherwise, I'll fix it tomorrow.

Reply:
> Actually, we can't simply pass x, because Datasets cannot be sorted

Okay, then the solution to that is to allow `sort!(d::Dataset)` which I'll do now quickly in DelayEmbeddings.jl and publish a release.

> no conversions to float are needed in dispersion_histogram (if dividing by a float). However, I'm not sure what the performance benefit is (if any).

This is a beyond-than-miniscule hit to performance and it is not worth the complication added to the source code.

If you can revert the code here and in the meantime I'll make `sort!` working!

Reply:
> Okay, then the solution to that is to allow sort!(d::Dataset) which I'll do now quickly in DelayEmbeddings.jl and publish a release.

Excellent.

> This is a beyond-than-miniscule hit to performance and it is not worth the complication added to the source code.
> If you can revert the code here and in the meantime I'll make sort! working!

Ok, sure! I'll do it first thing tomorrow.

Reply:
https://github.com/JuliaRegistries/General/pull/69481